tests/PHPStan/Rules/PhpDoc/MethodConditionalReturnTypeRuleTest.php

```diff
@@ -70,4 +70,14 @@ public function testRule(): void
 		]);
 	}
 
+	public function testBug8284(): void
+	{
+		$this->analyse([__DIR__ . '/data/bug-8284.php'], [
+			[
+				'Conditional return type references unknown parameter $callable.',
```

Comment:
there is no `@return` tag on the `replaceCallback()` method so is this error message a bit misleading?
it looks like it's really a conditional parameter type?

Reply:
The feature is actually "conditonal types", not "conditional return types" - it was implemented as a first thing after we released "conditional return types".

Reply:
But for the sake of backward compatibility, let's leave it like this.